### PR TITLE
update to 0.26.1 release

### PR DESCRIPTION
**Pull request by oxxenix**
Update release + solve the issue #32

### CODE REVIEW

File: charts/sgx-device-plugin/templates/sgx.yaml [hunk 1]
```diff
@@ -10,7 +10,6 @@ metadata:
   annotations: {{ toYaml .Values.annotations | nindent 4 }}
 spec:
   image: "{{ .Values.image.hub }}/intel-sgx-plugin:{{ .Values.image.tag | default .Chart.AppVersion }}"
-  initImage: "{{ .Values.initImage.hub }}/intel-sgx-initcontainer:{{ .Values.initImage.tag | default .Chart.AppVersion }}"
```
Comment (mythi) [inline charts/sgx-device-plugin/templates/sgx.yaml:13]: I'd prefer we drop the initContainer starting from `v0.27.0` and we have not released this option yet.

File: charts/device-plugin-operator/templates/operator.yaml [hunk 2]
```diff
@@ -462,7 +462,7 @@ spec:
         - --logtostderr=true
         - --tls-cipher-suites=TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305
         - --v=10
-        image: gcr.io/kubebuilder/kube-rbac-proxy:v0.13.1
+        image: "{{ .Values.kube-rbac-proxy.image.hub }}/kube-rbac-proxy:{{ .Values.kube-rbac-proxy.image.tag }}"
```
Comment (mythi) [inline charts/device-plugin-operator/templates/operator.yaml:465]: I'm OK with this but there was another ask to get private registries supported too. Did you consider adding that?

File: charts/device-plugin-operator/templates/operator.yaml [hunk 3]
```diff
@@ -462,7 +470,11 @@ spec:
         - --logtostderr=true
         - --tls-cipher-suites=TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305
         - --v=10
-        image: gcr.io/kubebuilder/kube-rbac-proxy:v0.13.1
+        {{- if .Values.privateRegistry.registryUrl }}
+        image: "{{ .Values.privateRegistry.registryUrl }}/{{ .Values.kubeRbacProxy.image.hubRepo }}:{{ .Values.kubeRbacProxy.image.tag }}"
+        {{- else }}
+        image: "{{ .Values.kubeRbacProxy.image.hub }}/{{ .Values.kubeRbacProxy.image.hubRepo }}/kube-rbac-proxy:{{ .Values.kubeRbacProxy.image.tag }}"
```
Comment (mythi) [inline charts/device-plugin-operator/templates/operator.yaml:476]: mismatch with `kubeRbacProxy` vs `kube-rbac-proxy`?

Comment (tkatila) [reply charts/device-plugin-operator/templates/operator.yaml:476]: Do we need the "privateRegistryUrl"? User could just set `manager.image.hub` or `kubeRbacProxy.image.hub` as he/she pleases?

File: charts/device-plugin-operator/values.yaml [hunk 4]
```diff
@@ -1,4 +1,17 @@
...
+  image:
+    hub: ghcr.io
+    hubRepo: kubebuilder
+    tag: v0.14.1
+    pullPolicy: IfNotPresent
+
+privateRegistry:
+  registryUrl: ""
+  registryUser: ""
+  registrySecret: ""
```
Comment (tkatila) [inline charts/device-plugin-operator/values.yaml:17]: For these, you'd also need to include the secret creation. I'd be fine to leave it as a later improvement.

File: charts/device-plugin-operator/templates/operator.yaml (hunk 3 above)
Comment (mythi) [reply charts/device-plugin-operator/templates/operator.yaml:476]: @tkatila is right and I prefer that option. it follows the same pattern as with the CRs

**Comment by mythi** [comment]
@MustDie95 please take a look

File: charts/device-plugin-operator/templates/operator.yaml [hunk 5]
```diff
@@ -427,8 +437,12 @@ spec:
           valueFrom:
             fieldRef:
               fieldPath: metadata.namespace
-        image: "{{ .Values.image.hub }}/intel-deviceplugin-operator:{{ .Values.image.tag | default .Chart.AppVersion  }}"
-        imagePullPolicy: {{ .Values.image.pullPolicy }}
+      {{- if .Values.privateRegistry.registrySecret }}
+      imagePullSecrets:
+      - name: {{ .Values.privateRegistry.registrySecret }}
+      {{- end }}
+        image: "{{ .Values.manager.image.hub }}/intel-deviceplugin-operator:{{ .Values.manager.image.tag | default .Chart.AppVersion  }}"
```
Comment (tkatila) [inline charts/device-plugin-operator/templates/operator.yaml:444]: The indentation is wrong here. imagePullSecrets includes container image details like "image"
It should be:
```
imagePullSecrets:
  - name: foobar
containers:
  - name: operator
```

Comment (tkatila) [reply charts/device-plugin-operator/templates/operator.yaml:444]: Sorry, my example is also wrong. The point I was trying to say is that with the current patch, the container spec would be under the imagePullSecrets. And that's wrong.

File: charts/device-plugin-operator/templates/operator.yaml [hunk 6]
```diff
@@ -398,6 +398,16 @@ spec:
...
     control-plane: controller-manager
 ---
+{{- if .Values.privateRegistry.registrySecret }}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ .Values.privateRegistry.registrySecret }}
+type: kubernetes.io/dockerconfigjson
+data:
+  .dockerconfigjson: {"{\"auths\": {\"%s\": {\"auth\": \"%s\"}"}}" .Values.privateRegistry.registryUrl (printf "%s:%s" .Values.privateRegistry.registryUser .Values.privateRegistry.registryPasswd | b64enc) | b64enc }}
```
[the remark this answers is not in the document]
Comment (MustDie95) [reply charts/device-plugin-operator/templates/operator.yaml:408]: Sorry, extra line breaks

.dockerconfigjson: {{ printf "{"auths": {"%s": {"auth": "%s"}}}" .Values.privateRegistry.registryUrl (printf "%s:%s" .Values.privateRegistry.registryUser .Values.privateRegistry.registryPassword | b64enc) | b64enc }}

File: charts/device-plugin-operator/templates/operator.yaml (hunk 5 above)
Comment (MustDie95) [reply charts/device-plugin-operator/templates/operator.yaml:444]: This section
```
{{- if .Values.privateRegistry.registrySecret }}
imagePullSecrets:
  - name: {{ .Values.privateRegistry.registrySecret }}
{{-end }}
```
should be completely BEFORE "containters" section
e.g.

```
spec:
    {{- if .Values.privateRegistry.registrySecret }}
    imagePullSecrets:
      - name: {{ .Values.privateRegistry.registrySecret }}
    {{- end }}
    containers:
    - args:
      {{- if .Values.controllerExtraArgs }}
      {{- with .Values.controllerExtraArgs }}
      {{- tpl . $ | trim | nindent 8 }}
      {{- end }}
...
```

**Review by mythi — APPROVED** [review]
LGTM

File: charts/device-plugin-operator/templates/operator.yaml [hunk 7]
```diff
@@ -398,6 +398,16 @@ spec:
...
     control-plane: controller-manager
 ---
+{{- if .Values.privateRegistry.registrySecret }}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ .Values.privateRegistry.registrySecret }}
+type: kubernetes.io/dockerconfigjson
+data:
+  .dockerconfigjson: {{ printf "{\"auths\": {\"%s\": {\"auth\": \"%s\"}"}}" .Values.privateRegistry.registryUrl (printf "%s:%s" .Values.privateRegistry.registryUser .Values.privateRegistry.registrySecret | b64enc) | b64enc }}
```
Comment (MustDie95) [inline charts/device-plugin-operator/templates/operator.yaml:408]: extra double quote here => }"}}

should be

.dockerconfigjson: {{ printf "{\"auths\": {\"%s\": {\"auth\": \"%\s"}}}"

Comment (MustDie95) [reply charts/device-plugin-operator/templates/operator.yaml:408]: But with this extra double quote I have got the error:

image

btw: this extra double quote makes the number of quotes in this string odd (13), so we have unterminated string

Without this extra double quote all is fine:

image

File: charts/device-plugin-operator/templates/operator.yaml [hunk 8]
```diff
@@ -398,6 +398,16 @@ spec:
   selector:
     control-plane: controller-manager
 ---
+{{- if .Values.privateRegistry.registrySecret }}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ .Values.privateRegistry.registrySecret }}
```
Comment (tkatila) [inline charts/device-plugin-operator/templates/operator.yaml:405]: Maybe the name of the secret shouldn't be the secret itself? I guess someone could list secrets while they can't get them?

File: charts/device-plugin-operator/templates/operator.yaml (hunk 7 above)
Comment (MustDie95) [reply charts/device-plugin-operator/templates/operator.yaml]: @obaranov1 it works, thanks

File: charts/device-plugin-operator/templates/operator.yaml [hunk 9]
```diff
@@ -398,6 +398,16 @@ spec:
   selector:
     control-plane: controller-manager
 ---
+{{- if .Values.privateRegistry.registrySecret }}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ .Release.Name }}-operator-private-registry"
```
Comment (tkatila) [inline charts/device-plugin-operator/templates/operator.yaml:405]: Extra `"` at the end.

File: charts/device-plugin-operator/templates/operator.yaml [hunk 10]
```diff
@@ -415,6 +425,10 @@ spec:
       labels:
         control-plane: controller-manager
     spec:
+      {{- if .Values.privateRegistry.registrySecret }}
+      imagePullSecrets:
+      - name: {{ .Values.privateRegistry.registrySecret }}
```
Comment (tkatila) [inline charts/device-plugin-operator/templates/operator.yaml:430]: This should be `{{ .Release.Name }}-operator-private-registry`

File: charts/device-plugin-operator/templates/operator.yaml [hunk 11]
```diff
@@ -415,6 +425,10 @@ spec:
       labels:
         control-plane: controller-manager
     spec:
+      {{- if {{.Release.Name }}-operator-private-registry
```
Comment (tkatila) [inline charts/device-plugin-operator/templates/operator.yaml:428]: This line was fine as it was. Also the postfix `}}` is missing from this line.

Comment (tkatila) [reply charts/device-plugin-operator/templates/operator.yaml:428]: No. Like this:
```
   spec:
      {{- if .Values.privateRegistry.registrySecret }}
      imagePullSecrets:
      - name: {{ .Release.Name }}-operator-private-registry }}
      {{- end }}
```

**Review by tkatila — APPROVED** [review]
lgtm

**Review by poussa — APPROVED** [review]

**Pull request merged**